### ISSUE TITLE
Error when inserting values using subquery

### ISSUE THREAD

**Issue by Marco-GG** [hiddentao/squel#111]
Using the example showed in http://hiddentao.github.io/squel/#insert, the generated query output of squel differs with the one showed in the comments of the example.

The example states that the output will be: "INSERT INTO students (name, age, score, graduate, nickname, level) VALUES ('Thomas', 29, 90.2, TRUE, NULL, (SELECT MAX(level) FROM levels))"

But I'm getting this output: INSERT INTO students (name, age, score, graduate, nickname, level) VALUES (?, ?, ?, ?, ?, SELECT MAX(level) FROM levels)

The lack of the parenthesis in the subquery gives an syntax error in MariaDB.

I've detected this issue in squel version 3.10.0.

Regards
Marco

**Comment by hiddentao** [comment]
I am no longer seeing this error. And the subquery appears to be in parentheses, as required.

**Comment by Marco-GG** [comment]
Did you try building a parameterized query? I'm sorry, I should clarify before that I'm getting this problem only with the parameterized version of the query.

Using squel 3.10 this are the outputs I get:
```javascript
> squel.insert().into("students").set("name", "Thomas").set("level", squel.select().field('MAX(level)').from('levels')).toString()

**'INSERT INTO students (name, level) VALUES (\'Thomas\', (SELECT MAX(level) FROM levels))'** //OK!
```
But with toParam:
```javascript
> squel.insert().into("students").set("name", "Thomas").set("level", squel.select().field('MAX(level)').from('levels')).toParam()

**{ text: 'INSERT INTO students (name, level) VALUES (?, SELECT MAX(level) FROM levels)',
  values: [ 'Thomas' ] }** // Parenthesis lost!
```
In the subquery test case (insert.test.coffee:125) the expected query doesn't have parenthesis, so the question is: The behavior of toParam with a subquery is the expected behavior?

**Comment by MayhemYDG** [comment]
Have this issue too with postgres.
My "fix" was to do:
```js
var count = squel.select().field('count(*) + 1').from('table');
var statement = squel.insert()
  .into('table')
  .setFields({...})
  .toParam();

statement.text = statement.text
  .replace('(', '(num, ')
  .replace('VALUES (', `VALUES ((${count}), `);
```

**Comment by MayhemYDG** [comment]
Also there is a reference to `.fromQuery` here: https://hiddentao.github.io/squel/#insert
Which I assume is an error since it's not in the API docs and it's not usable.

**Comment by hiddentao** [comment]
`fromQuery()` is now available in 3.11.0.
